### PR TITLE
how to fix a misused meshkit logger

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
Signed-off-by: Rudraksh Pareek <rudrakshpareek3601@gmail.com>

**Description**

This PR fixes an instance of misused meshkit logger

**Notes for Reviewers**
**We need to replace this everywhere to prevent frequent panics which are hiding in plain sight but are noticed only when an error actually occurs.**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by sayantan1413 — COMMENTED** [review]
Yes, I agree @DelusionalOptimist it makes sense to me. I think we have skipped some of this and so we need to fix it.

File: istio/sample_apps.go [hunk 1]
```diff
@@ -124,9 +124,23 @@ func (istio *Istio) LoadNamespaceToMesh(namespace string, remove bool) error {
 		delete(ns.ObjectMeta.Labels, "istio-injection")
 	}
 
+	// How to use meshkit errors which are logged using meshkit logger
+	//
+	// The LoadNamespaceToMesh() function is called in istio/istio.go and the
+	// error returned by this is logged through the meshkit logger
+	//
+	// The below function returns Go's built in error type but we can't
```
Comment (Mr-DG-Wick) [inline istio/sample_apps.go:132]:
```suggestion
	// The below function returns Go's built-in error type but we can't
```

**Review by Mr-DG-Wick — CHANGES_REQUESTED** [review]

File: istio/sample_apps.go (hunk 1 above)
Comment (DelusionalOptimist) [reply istio/sample_apps.go:132]: :heavy_check_mark:

**Comment by leecalcote** [comment]
@Shreyas220 will you add a note on this to https://docs.meshery.io/project/contributing/contributing-error?

**Comment by leecalcote** [comment]
@Shreyas220 will you examine other adapters for this same need to change the error logging?

**Comment by Shreyas220** [comment]
> @Shreyas220 will you add a note on this to https://docs.meshery.io/project/contributing/contributing-error?

@DelusionalOptimist has added the docs already in https://github.com/meshery/meshery/pull/4594

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**